### PR TITLE
fix(types): changed NodeJS.ProcessEnv type to Record in node-config-provider

### PR DESCRIPTION
**Pull request by froggydood**
node-config-provider

### Description
Changed the NodeJS.ProcessEnv type (in the node-config-provider package), to a Record<string, string | undefined> type. This removes the dependency on @types/node for the node-config-provider. This means that when the types are compiled there is no reliance on it and it can be therefore used in more non-node environments, for example cloudflare worker environments, without any type mismatches or errrors.

### Testing
Changed the type of `envVarGetter` variable in the `fromEnv.spec.ts` file, to match the new type file. Because its only a type change, there is only minimal test changes.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by froggydood** [comment]
Created issue [#4329](https://github.com/aws/aws-sdk-js-v3/issues/4329)

File: packages/node-config-provider/src/configLoader.spec.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ describe("loadConfig", () => {
     (fromSharedConfigFiles as jest.Mock).mockReturnValueOnce(mockFromSharedConfigFilesReturn);
     const mockFromStatic = "mockFromStatic";
     (fromStatic as jest.Mock).mockReturnValueOnce(mockFromStatic);
-    const envVarSelector = (env: NodeJS.ProcessEnv) => env["AWS_CONFIG_FOO"];
+    const envVarSelector = (env: Record<string, string | undefined>) => env["AWS_CONFIG_FOO"];
```
Comment (yenfryherrerafeliz) [inline packages/node-config-provider/src/configLoader.spec.ts:28]: I agree with the suggested change. Would you be able to please put some comments about why this change is being done?. For example, you can mention that NodeJS.ProcessEnv causes types error on non node environments.

Comment (froggydood) [reply packages/node-config-provider/src/configLoader.spec.ts:29]: I did another commit, and added comments to the code. I think that was what you meant, or did you mean comments to the PR?

Comment (yenfryherrerafeliz) [reply packages/node-config-provider/src/configLoader.spec.ts:29]: Yeah, that is what I meant.

Thanks!

**Review by yenfryherrerafeliz — APPROVED** [review]

**Review by kuhe — APPROVED** [review]
